### ISSUE TITLE
Writer concept is not flexible enough

### ISSUE THREAD

**Issue by Ingrater** [lodo1995/experimental.xml#37]
I have a xml document that looks like this

```
<?xml version="1.0" encoding="UTF-8"?>
<root>
  <source>
    <Item>Text1</Item>
    <Item>Text2</Item>
  <source>
<root>
```

The current writer concept does not allow formatting the document in this style. "afterNode" will be called always, even after text nodes. There is no context given to "afterNode" so its impossible to decide whether or not to place a newline. The same goes for the indentation. The writer concept should be extended to allow for this style of formatting (as its very common)

**Comment by lodo1995** [comment]
You are right. The hooks are not complete. I need to find the best balance between having the maximum customization and keeping the number of hooks reasonable.

**Comment by wilzbach** [comment]
This issue was moved to dlang-community/experimental.xml#13
